sdks/go/pkg/beam/core/runtime/xlangx/expansionx/download_test.go

```diff
@@ -0,0 +1,101 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expansionx
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestGetURLForBeamJar(t *testing.T) {
+	tests := []struct {
+		name        string
+		target      string
+		version     string
+		expectedURL url
+	}{
+		{
+			"base",
+			":sdks:java:fake",
+			"VERSION",
+			"https://repo.maven.apache.org/maven2/org/apache/beam/beam-sdks-java-fake/VERSION/beam-sdks-java-fake-VERSION.jar",
+		},
+		{
+			"versioned",
+			":sdks:java:fake",
+			"1.2.3",
+			"https://repo.maven.apache.org/maven2/org/apache/beam/beam-sdks-java-fake/1.2.3/beam-sdks-java-fake-1.2.3.jar",
+		},
+	}
+	for _, test := range tests {
+		j := defaultJarGetter
```

Comment:
Use the construction function for the default here, rather than testing against the default. This way lies non-hermetic tests.

If you have a concern, you can always have one test check that the defaultJarGetter is equal to the one in the construction function.